Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp

```diff
@@ -348,55 +358,75 @@ RefPtr<DisplayRefreshMonitor> LayerTreeHost::createDisplayRefreshMonitor(Platfor
 
 void LayerTreeHost::didFlushRootLayer(const FloatRect& visibleContentRect)
 {
+    ASSERT(isMainRunLoop());
+
     // Because our view-relative overlay root layer is not attached to the FrameView's GraphicsLayer tree, we need to flush it manually.
     if (m_viewOverlayRootLayer)
         m_viewOverlayRootLayer->flushCompositingState(visibleContentRect);
 }
 
 void LayerTreeHost::commitSceneState(const RefPtr<Nicosia::Scene>& state)
 {
+    ASSERT(isMainRunLoop());
+
     m_isWaitingForRenderer = true;
     m_compositor->updateSceneState(state, ++m_compositionRequestID);
 }
 
 void LayerTreeHost::updateScene()
 {
+    ASSERT(isMainRunLoop());
+
     m_compositor->updateScene();
 }
 
 void LayerTreeHost::frameComplete()
 {
+    ASSERT(isMainRunLoop());
+
```

Comment:
This one is not correct, at least for GTK and WPE with the new API, AcceleratedSurfaceDMABuf receives the IPC messages in the compositing thread, and this is called from FrameDone message callback.

Reply:
Ah, yeah, I need to validate with the proper compilation, thanks for the review!